src/components/application_manager/test/rpc_passing_handler_test.cc

```diff
@@ -0,0 +1,447 @@
+    app_service.mobile_service = true;
+    app_service.default_service = false;
+
+    smart_objects::SmartObject record(smart_objects::SmartType::SmartType_Map);
+    record[am::strings::service_id] = service_id;
+    // record[am::strings::service_manifest][am::strings::service_name] =
+    // service_name;
+    record[am::strings::service_manifest][am::strings::service_type] =
+        service_type;
+    record[am::strings::service_manifest][am::strings::allow_app_consumers] =
+        true;
+    record[am::strings::service_manifest][am::strings::handled_rpcs] =
+        smart_objects::SmartObject(smart_objects::SmartType::SmartType_Array);
+    record[am::strings::service_manifest][am::strings::handled_rpcs][0] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::service_published] = true;
+    record[am::strings::service_active] = true;
+    app_service.record = record;
+
+    return app_service;
+  }
+
+  smart_objects::SmartObject CreatePassThroughRequest(uint32_t connection_key,
+                                                      int32_t correlation_id) {
+    smart_objects::SmartObject record(smart_objects::SmartType::SmartType_Map);
+    record[am::strings::params][am::strings::connection_key] = connection_key;
+    record[am::strings::params][am::strings::correlation_id] = correlation_id;
+    record[am::strings::params][am::strings::function_id] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::params][am::strings::message_type] =
+        am::MessageType::kRequest;
+    record[am::strings::msg_params][am::strings::latitude_degrees] = 50;
+    record[am::strings::msg_params][am::strings::longitude_degrees] = 50;
+    record[am::strings::msg_params][am::strings::location_name] =
+        "TestLocation";
+
+    return record;
+  }
+
+  smart_objects::SmartObject CreatePassThroughResponse(
+      uint32_t connection_key,
+      int32_t correlation_id,
+      const std::string& result_code,
+      bool success,
+      std::string info = std::string()) {
+    smart_objects::SmartObject record(smart_objects::SmartType::SmartType_Map);
+    record[am::strings::params][am::strings::connection_key] = connection_key;
+    record[am::strings::params][am::strings::correlation_id] = correlation_id;
+    record[am::strings::params][am::strings::function_id] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::params][am::strings::message_type] =
+        am::MessageType::kResponse;
+    record[am::strings::msg_params][am::strings::result_code] = result_code;
+    record[am::strings::msg_params][am::strings::success] = success;
+    record[am::strings::msg_params][am::strings::info] = info;
+
+    return record;
+  }
+
+  void SendRequestToASP(const PassthroughParams& params, uint32_t timeout = 0) {
+    if (timeout == 0) {
+      // If timeout argument is not sepcified
+      timeout = kRPCPassthroughTimeout;
+    }
+
+    {
+      InSequence dummy;
```

Comment:
Is it possible to avoid using `InSequence`? I would assume this means that the functions need to be called in order, which I imagine would lead to fragile tests.

Reply:
Resolved in https://github.com/smartdevicelink/sdl_core/pull/2912/commits/d194dfd9e425d0539a1c4a487eaf5dd389cd623b